src/app/page.js

```diff
@@ -63,6 +63,45 @@ function Home() {
     newItems.splice(index, 1);
     setItems(newItems);
   };
+
+  const handleTabRename = (tab, event) => {
+    const index = items.findIndex((i, k) => k === tab);
+    const currentTab = event.target;
+    let currentName = currentTab.textContent;
+
+    // Creating Form/Input to collect new tab name
+    let form = document.createElement('form');
+    currentTab.replaceChildren(form);
+    let inputField = document.createElement('input');
+    inputField.value = currentName;
+    form.appendChild(inputField);
+    currentTab.firstChild[0].focus();
+    currentTab.firstChild[0].select();
+
+    // Add Submit Event
+    currentTab.firstChild.addEventListener('submit', tabRenameHandler)
+
+    // Rename Handler
+    function tabRenameHandler(e){
+      e.preventDefault();
+      let newName = e.target[0].value;
+
+      const newItems = items.map((item, idx) => ({
+        ...item,
+        title: idx === index ? newName : item.title,
+      }));
+      setItems(newItems);
+      
+      currentTab.replaceChildren(newName);
+    }
+
+    // On Click Away from Tab Renaming
+    window.addEventListener('click', () => {
```

Comment:
This code somehow makes the tab rename reset.

Reply:
The rename form is suppose to take a submit action i.e. a click on enter/return key to set/save the new name.

This listener is meant to restore the last set/saved name in the case where a user clicks away from the input form (i.e. an action that means they no longer wish to rename)

Reply:
I have since discovered this to be the problem.

Considering that my main intention was to keep whatever value the tab currently holds in the form when it loses focus, adding a click event listener to the `window` object was an incorrect approach as it restores the name to default tab name when a click event occurs anywhere, anytime outside the tab.

I will replace this with a `focusout` event on the tab form itself, to keep the current value as the tab name.